tests/pipeline/test_quarters_estimates.py

```diff
@@ -0,0 +1,898 @@
+                # Filter out estimates we don't know about yet.
+                ts_eligible_estimates = ts_sorted_estimates[
+                    ts_sorted_estimates[TS_FIELD_NAME] <= comparable_date
+                ]
+                # If there are estimates we know about:
+                if not ts_eligible_estimates.empty:
+                    # Determine the last piece of information we know about
+                    # for q1 and q2. This takes advantage of the fact that we
+                    # only have 2 quarters in the test data.
+                    q1_knowledge = ts_eligible_estimates[
+                        ts_eligible_estimates[FISCAL_QUARTER_FIELD_NAME] == 1
+                    ]
+                    q2_knowledge = ts_eligible_estimates[
+                        ts_eligible_estimates[FISCAL_QUARTER_FIELD_NAME] == 2
+                    ]
+                    expected_estimate = self.get_expected_estimate(
+                        q1_knowledge,
+                        q2_knowledge,
+                        comparable_date,
+                    )
+                    if not expected_estimate.empty:
+                        for colname in sid_estimates.columns:
+                            expected_value = expected_estimate[colname]
+                            computed_value = sid_estimates.iloc[i][colname]
+                            assert_equal(expected_value, computed_value)
+                    else:
+                        # There are no eligible 'next' estimates on this day;
+                        #  everything should be null.
+                        assert_true(sid_estimates.iloc[i].isnull().all())
+                else:
+                    # We don't know about any estimates on this day;
+                    # everything should be null.
+                    assert_true(sid_estimates.iloc[i].isnull().all())
+
+
+class NextEstimate(WithEstimatesT0, ZiplineTestCase):
+    @classmethod
+    def make_loader(cls, events, columns):
+        return NextQuartersEstimatesLoader(events, columns)
+
+    def get_expected_estimate(self,
+                              q1_knowledge,
+                              q2_knowledge,
+                              comparable_date):
+        # If our latest knowledge of q1 is that the release is
+        # happening on this simulation date or later, then that's
+        # the estimate we want to use.
+        if (not q1_knowledge.empty and
+            q1_knowledge.iloc[-1][EVENT_DATE_FIELD_NAME] >=
+                comparable_date):
+            return q1_knowledge.iloc[-1]
+        # If q1 has already happened or we don't know about it
+        # yet and our latest knowledge indicates that q2 hasn't
+        # happened yet, then that's the estimate we want to use.
+        elif (not q2_knowledge.empty and
+              q2_knowledge.iloc[-1][EVENT_DATE_FIELD_NAME] >=
+                comparable_date):
+            return q2_knowledge.iloc[-1]
+        return pd.DataFrame()
```

Comment:
instead of checking for an empty frame, could we just return `None` and check for that?